src/Service.Tests/SqlTests/RestApiTests/Patch/MsSqlPatchApiTests.cs

```diff
@@ -190,6 +208,78 @@ public async Task PatchOneViewBadRequestTest()
             await base.PatchOneViewBadRequestTest(expectedErrorMessage);
         }
 
+        /// <summary>
+        /// Test to validate that even when an update DML trigger is enabled on a table, we still return the
+        /// latest data as it is present after the trigger gets executed. Whenever an update DML trigger is enabled,
+        /// we use a subsequent SELECT query to get the data instead of using OUTPUT clause.
+        /// </summary>
+        [TestMethod]
+        public async Task PatchOneUpdateInTableWithUpdateTrigger()
+        {
+            // Validate that PATCH operation (resulting in update) succeeds when an update DML trigger is configured for a table
+            // with autogenerated primary key. The trigger updates the salary to max(0,min(150,salary)).
+            string requestBody = @"
+            {
+                ""salary"": -9
+            }";
+
+            await SetupAndRunRestApiTest(
+                primaryKeyRoute: "id/1/u_id/2",
```

Comment:
does this prevent updating multiple columns when more than one record has `u_id == 2`?

Reply:
Not clear with the question. What do we need to prevent here? To uniquely identify a record, we are supplying the value for each of the PK field.

Reply:
i think i cleared up  my understand by re-reading mssqlquerybuilder implementation/new comments you added. I was confused about the implications of having auto-gen pk in an added where clause on the main table and the non-autogen pk's existing in the temp table.